modules/cfe_testcase/src/msg_api_test.c

```diff
@@ -0,0 +1,231 @@
+**      limitations under the License.
+**
+** File: msg_api_test.c
+**
+** Purpose:
+**   Functional test of cFE Message header APIs.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+#include <string.h>
+
+#define UtAssert_SUCCESS(status) UtAssert_INT32_EQ(status, CFE_SUCCESS)
+
+ /* merged all test into one function to avoid creating new messages for each test */
+void TestMsgApi(void)
+{
+    UtPrintf("Testing: CFE_MSG_Init, CFE_MSG_GetSize, CFE_MSG_SetSize, CFE_MSG_GetType, "
+             "CFE_MSG_SetType, CFE_MSG_GetHeaderVersion, CFE_MSG_SetHeaderVersion, "
+             "CFE_MSG_GetHasSecondaryHeader, CFE_MSG_SetHasSecondaryHeader,  CFE_MSG_GetApId, "
+             "CFE_MSG_SetApId, CFE_MSG_GetSegmentationFlag, CFE_MSG_SetSegmentationFlag, "
+             "CFE_MSG_GetSequenceCount, CFE_MSG_SetSequenceCount, CFE_MSG_GetNextSequenceCount, "
+             "CFE_MSG_GetEDSVersion, CFE_MSG_SetEDSVersion, CFE_MSG_GetEndian, CFE_MSG_SetEndian, "
+             "CFE_MSG_GetPlaybackFlag, CFE_MSG_SetPlaybackFlag, CFE_MSG_GetSubsystem, "
+             "CFE_MSG_SetSubsystem, CFE_MSG_GetSystem, CFE_MSG_SetSystem, CFE_MSG_GenerateChecksum, "
+             "CFE_MSG_ValidateChecksum, CFE_MSG_SetFcnCode, CFE_MSG_GetFcnCode, "
+             "CFE_MSG_GetMsgTime, CFE_MSG_SetMsgTime ");
+
+    /* declare local vars */
+    CFE_MSG_CommandHeader_t    cmd;
+    CFE_MSG_Size_t             size;
+    CFE_MSG_Type_t             type;
+    CFE_SB_MsgId_Atom_t        msgId;
+    CFE_MSG_HeaderVersion_t    hdrVer;
+    CFE_MSG_ApId_t             appId;
+    CFE_MSG_SegmentationFlag_t segFlag;
+    CFE_MSG_SequenceCount_t    seqCnt;
+    CFE_MSG_EDSVersion_t       edsVersion;
+    CFE_MSG_Endian_t           endian;
+    CFE_MSG_PlaybackFlag_t     playbackFlag;
+    CFE_MSG_Subsystem_t        subSystem;
+    CFE_MSG_System_t           system;
+    CFE_MSG_FcnCode_t          fcnCode;
+    CFE_TIME_SysTime_t         msgTime;
+    bool                       hasSecondaryHdr;
+    /* bool                       isValid; */
+
+    memset(&cmd, 0xFF, sizeof(cmd));
+    msgId = 0;
+
+    /* test msg-init */
+    UtAssert_INT32_EQ(CFE_MSG_Init(NULL, CFE_SB_ValueToMsgId(0), sizeof(cmd)), CFE_MSG_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_MSG_Init(&cmd.Msg, CFE_SB_ValueToMsgId(0), 0), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_Init(&cmd.Msg, CFE_SB_ValueToMsgId(msgId), sizeof(cmd)));
+
+    /* test set-msg-size */
+    UtAssert_SUCCESS(CFE_MSG_SetSize(&cmd.Msg, 12u));
+
+    /* test get-msg-size */
+    UtAssert_SUCCESS(CFE_MSG_GetSize(&cmd.Msg, &size));
+    UtAssert_True(size == 12u, "CFE_MSG_SetSize => size = 12u");
+
+    /* test get-type */
+    UtAssert_SUCCESS(CFE_MSG_GetType(&cmd.Msg, &type));
+    /* type has to be ...Type_Cmd because it created within a command */
+    UtAssert_INT32_EQ(type, CFE_MSG_Type_Cmd);
```

Comment:
Default behavior is implementation dependent, preference would be to set the type explicitly  for command (and check it) like you do for Tlm.